.github/actions/build-push-artifacts/action.yml

```diff
@@ -0,0 +1,63 @@
+name: 'Builds and Push Artifacts'
+description: 'Builds and pushes a Docker image, S3 env vars file, and dispatches a Twistlock workflow'
+inputs:
+  environment:
+    required: true
+    default: "dev"
+    type: string
+  ref:
+    description: 'Git reference (commit SHA, branch, or tag) to build'
+    required: true
+  aws-access-key-id:
+    description: 'AWS Access Key ID'
+    required: true
+  aws-secret-access-key:
+    description: 'AWS Secret Access Key'
+    required: true
+  role-to-assume:
+    description: 'AWS role to assume'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.ref }}
+
+    - name: Configure VAEC AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v4.0.2
+      with:
+        aws-access-key-id: ${{ inputs.aws-access-key-id }}
+        aws-secret-access-key: ${{ inputs.aws-secret-access-key }}
+        aws-region: us-gov-west-1
+        role-to-assume: ${{ inputs.role-to-assume }}
+        role-skip-session-tagging: true
+        role-duration-seconds: 900
```

Comment:
Actions do not have access to repo secrets, so we need to pass them in. We already do this (like https://github.com/department-of-veterans-affairs/notification-api/blob/main/.github/workflows/twistlock.yml#L21 for example).